client/rpc/src/dev/mod.rs

```diff
@@ -104,7 +104,7 @@ where
 			.execute_block(&BlockId::Hash(parent_header.hash()), block)
 			.map_err(|_| Error::BlockExecutionFailed)?;
 		let witness = runtime_api
-			.extract_proof()
+			.extract_proof(&BlockId::Hash(parent_header.hash()))
```

Comment:
```suggestion
			.extract_proof(&BlockId::Hash(block.header().parent_hash()))
```
Save on hash computation.